### PR TITLE
fix: use JSX directly in routes

### PR DESCRIPTION
**Pull request by danielgent**
Another PR where the diff is too large :see_no_evil:

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -10,10 +10,7 @@ import loggerMiddleware from "./middlewares/logger";
 import renderOauthRedirectHtml from "./routes/oauth2-redirect";
 import { validateUrl } from "./utils/validate-redirect-url";
 import { Var } from "./types/Var";
-import { renderReactThing } from "./utils/reactdemo";
-import validatePassword from "./utils/validatePassword";
-import { getUserByEmailAndProvider } from "./utils/users";
-import { getClient } from "./services/clients";
+import { getResetPassword, postResetPassword } from "./routes/tsx/routes";
```
Comment (danielgent) [inline src/app.ts:13]: as this file is not `app.tsx` we cannot use TSX directly in here

That wouldn't scale though anyway! This is a pretty crude way of handling routes :smile:

File: src/routes/tsx/routes.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  const session = await env.data.universalLoginSessions.get(state);
+  if (!session) {
+    throw new HTTPException(400, { message: "Session not found" });
+  }
+
+  if (!validatePassword(password)) {
+    return ctx.html(
+      <ResetPasswordPage error="Password does not meet the requirements" />,
+      400,
+    );
```
Comment (danielgent) [inline src/routes/tsx/routes.tsx:77]: This is the only change here that isn't a copy paste

Rathern than do
```
   return renderReactThing(
        ctx,
        "Password does not meet the requirements",
        400,
      );
```
      
I'm instead using JSX directly


This is another PR that should be split up really due to the diffs caused by moving routes around

*Just say the word and I'll do it* :joy:

File: src/utils/reactdemo.tsx [hunk 3]
```diff
@@ -15,7 +15,7 @@ const Button: FC<{}> = ({ children }) => {
   );
 };
 
-const ResetPasswordPage: FC<{ error?: string }> = ({ error }) => {
+export const ResetPasswordPage: FC<{ error?: string }> = ({ error }) => {
```
Comment (danielgent) [inline src/utils/reactdemo.tsx:18]: these should all be in their own component file now in the components folder I made...


I'll do this now on another PR

Comment (danielgent) [reply src/utils/reactdemo.tsx:18]: https://github.com/sesamyab/auth/pull/579 :eyes:  that looks more like React!

**Review by markusahlstrand — APPROVED** [review]
Looks good

**Pull request merged**